src/qibo/backends/__init__.py

```diff
@@ -22,34 +35,24 @@ def __init__(self):
                                              "Please install it using "
                                              "`pip install numpy`.")
 
-        # check if tensorflow is installed and use it as default backend.
-        if self.check_availability("tensorflow"):
-            os.environ["TF_CPP_MIN_LOG_LEVEL"] = str(config.TF_LOG_LEVEL)
-            import tensorflow as tf  # pylint: disable=E0401
-            if tf.__version__ < TF_MIN_VERSION:  # pragma: no cover
-                raise_error(
-                    RuntimeError, f"TensorFlow version not supported, minimum is {TF_MIN_VERSION}.")
-            from qibo.backends.tensorflow import TensorflowBackend
-            self.available_backends["tensorflow"] = TensorflowBackend
-            active_backend = "tensorflow"
-            if self.check_availability("qibotf"):
-                from qibo.backends.tensorflow import TensorflowCustomBackend
-                self.available_backends["qibotf"] = TensorflowCustomBackend
-                active_backend = "qibotf"
-
-        # check if qibojit is installed and use it as default backend.
-        if self.check_availability("qibojit"): # pragma: no cover
-            # qibojit backend is not tested until `qibojit` is available
-            from qibo.backends.jit import JITCustomBackend
-            self.available_backends["qibojit"] = JITCustomBackend
-            active_backend = "qibojit"
-
-        # check if IcarusQ is installed
-        if self.check_availability("qiboicarusq"): # pragma: no cover
-            # hardware backend is not tested until `qiboicarusq` is available
-            from qibo.backends.hardware import IcarusQBackend
-            self.available_backends["icarusq"] = IcarusQBackend
-            self.hardware_backends["icarusq"] = IcarusQBackend
+        for backend in profile.get('backends'):
+            name = backend.get('name')
+            if self.check_availability(name):
+                if name == 'tensorflow':
+                    os.environ["TF_CPP_MIN_LOG_LEVEL"] = str(
+                        config.TF_LOG_LEVEL)
+                    import tensorflow as tf  # pylint: disable=E0401
+                    if tf.__version__ < TF_MIN_VERSION:  # pragma: no cover
+                        raise_error(
+                            RuntimeError, f"TensorFlow version not supported, minimum is {TF_MIN_VERSION}.")
+                import importlib
+                custom_backend = getattr(importlib.import_module(
+                    backend.get('from')), backend.get('class'))
+                self.available_backends[name] = custom_backend
+                if backend.get('is_hardware', False):  # pragma: no cover
+                    self.hardware_backends['qiboicarusq'] = custom_backend
```

Comment:
I believe this should be
```suggestion
                    self.hardware_backends[name] = custom_backend
```
so that it works for arbitrary hardware backends.

Reply:
Absolutely.